code/modules/mob/living/simple_animal/friendly/spiderbot.dm

```diff
@@ -119,25 +119,43 @@
 			return
 	else if(O.GetID())
 		if (!mmi)
-			to_chat(user, "<span class='danger'>There's no reason to swipe your ID - \the [src] has no brain to remove.</span>")
+			to_chat(user, "<span class='danger'>There's no reason to swipe your ID - \the [src] has no brain.</span>")
 			return 0
 
 		var/obj/item/card/id/id_card = O.GetID()
 
 		if(!istype(id_card))
 			return 0
 
-		if(access_robotics in id_card.access)
-			to_chat(user, "<span class='notice'>You swipe your access card and pop the brain out of \the [src].</span>")
-			eject_brain()
-			if(held_item)
-				held_item.forceMove(src.loc)
-				held_item = null
-			return 1
-		else
-			to_chat(user, "<span class='danger'>You swipe your card with no effect.</span>")
-			return 0
+		var/choice = input(user, "Would you like to eject the brain or sync access?", "Swipe Mode") as null|anything in list("Eject", "Sync")
+		if(!choice)
+			return
 
+		switch(choice)
+			if("Eject")
+				if(!use_check_and_message(user))
+					if(access_robotics in id_card.access)
+						to_chat(user, "<span class='notice'>You swipe your access card and pop the brain out of \the [src].</span>")
+						eject_brain()
+						if(held_item)
+							held_item.forceMove(src.loc)
+							held_item = null
+						return 1
+					else
+						to_chat(user, "<span class='danger'>You swipe your card with no effect.</span>")
+						return 0
+				else
+					to_chat(user, "<span class='danger'>You are unable to swipe!</span>")
+					return 0
+			if("Sync")
+				if(!use_check_and_message(user))
+					internal_id.access.Cut()
+					internal_id.access = id_card.access.Copy()
+					to_chat(user, SPAN_NOTICE("Access synced with [src]"))
+					return 1
+				else
+					to_chat(user, "<span class='danger'>You are unable to swipe!</span>")
+					return 0
```

Comment:
`use_check_and_message `is already informing the `user` why it fails.
So there is no need for that extra message.

This should work -> But should be tested since its made using the github suggestion feature

```suggestion
			if("Eject")
				if(use_check_and_message(user))
					return 0
				if(access_robotics in id_card.access)
					to_chat(user, "<span class='notice'>You swipe your access card and pop the brain out of \the [src].</span>")
					eject_brain()
					if(held_item)
						held_item.forceMove(src.loc)
						held_item = null
					return 1
				else
					to_chat(user, "<span class='danger'>You swipe your card with no effect.</span>")
					return 0
			if("Sync")
				if(use_check_and_message(user))
					return 0

				internal_id.access.Cut()
				internal_id.access = id_card.access.Copy()
				to_chat(user, SPAN_NOTICE("Access synced with [src]"))
				return 1
```